File: jeonghye-choi/42840.js
//---------------------------------------------------------------------------
// Title:         프로그래머스 완전탐색 모의고사
// Author:        jeonghye-choi
// Last Updated:  2022.2.22
// Link:          https://programmers.co.kr/learn/courses/30/lessons/42840
//---------------------------------------------------------------------------

function solution(answers) {
  let count = [0, 0, 0];
  let result = [];

  const student1 = [1, 2, 3, 4, 5]; // 5개 반복
  const student2 = [2, 1, 2, 3, 2, 4, 2, 5]; //8개 반복
  const student3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]; // 10개 반복

  answers.forEach((answer, index) => {
    if (student1[index % 5] === answer) {
      count[0] += 1;
    }
    if (student2[index % 8] === answer) {
      count[1] += 1;
    }
    if (student3[index % 10] === answer) {
      count[2] += 1;
    }
  });

  var max = 0;
  count.forEach((value, index) => {
    if (value > max) {
      max = value;
      result = [index + 1];
    } else if (value === max) {
      result.push(index + 1);
    }
  });

  return result;
}

const answers = [1, 3, 2, 4, 2];
solution(answers);

File: cmsong111/42840.py
def solution(answers):
    scores = [0,0,0]
    answer = []
    count = len(answers)

    s1 = [1,2,3,4,5]*(count//5+1)  #5
    s2 = [2,1,2,3,2,4,2,5]*(count//8+1) #8 
    s3 = [3,3,1,1,2,2,4,4,5,5]*(count//10+1) #10
    
    
    for i in range(count): 
        if answers[i] == s1[i]:
            scores[0] +=1
        if answers[i] == s2[i]:
            scores[1] +=1
        if answers[i] == s3[i]:
            scores[2] +=1
    max_score = max(scores)

    print("학생들 점수는: ",scores)

    for i in range(3):
        if max_score == scores[i]:
            answer.append(i+1)

    print("정답: ",answer)
    return answer


#https://programmers.co.kr/learn/courses/30/lessons/42840

answers = [1,2,3,4,5]
solution(answers)

print()

answers = [1,3,2,4,2]
solution(answers)